reliability/single-fit/rum/src/test/kotlin/com/datadog/android/rum/integration/RumResourceInputStreamTest.kt

```diff
@@ -0,0 +1,181 @@
+            }
+            .hasRumEvent(index = 1) {
+                hasService(stubSdkCore.getDatadogContext().service)
+                hasApplicationId(fakeApplicationId)
+                hasSessionType("user")
+                hasSource("android")
+                hasType("resource")
+                hasViewName(viewName)
+                hasResourceUrl(resourceUrl)
+            }
+            .hasRumEvent(index = 2) {
+                hasService(stubSdkCore.getDatadogContext().service)
+                hasApplicationId(fakeApplicationId)
+                hasSessionType("user")
+                hasSource("android")
+                hasType("view")
+                hasViewName(viewName)
+                hasResourceCount(1)
+            }
+    }
+
+    @RepeatedTest(4)
+    fun `M report RUM Error W asRumResource() + read()`(
+        @StringForgery viewKey: String,
+        @StringForgery viewName: String,
+        @StringForgery resourceUrl: String,
+        @Forgery error: Throwable
+    ) {
+        // Given
+        GlobalRumMonitor.get(stubSdkCore).startView(viewKey, viewName)
+        val inputStream: InputStream = mock()
+        val rumResourceInputStream = RumResourceInputStream(inputStream, resourceUrl, stubSdkCore)
+        whenever(inputStream.read()) doThrow error
+
+        // When
+        var forwardedError: Throwable? = null
+        try {
+            rumResourceInputStream.read()
+        } catch (e: Throwable) {
+            forwardedError = e
+        }
+
+        // Then
+        assertThat(forwardedError).isEqualTo(error)
+        val eventsWritten = stubSdkCore.eventsWritten(Feature.RUM_FEATURE_NAME)
+        assertThat(eventsWritten).hasSize(3)
+            .hasRumEvent(index = 0) {
+                hasService(stubSdkCore.getDatadogContext().service)
+                hasApplicationId(fakeApplicationId)
+                hasSessionType("user")
+                hasSource("android")
+                hasType("view")
+                hasViewName(viewName)
+                hasResourceCount(0)
+                hasErrorCount(0)
+            }
+            .hasRumEvent(index = 1) {
+                hasService(stubSdkCore.getDatadogContext().service)
+                hasApplicationId(fakeApplicationId)
+                hasSessionType("user")
+                hasSource("android")
+                hasType("error")
+                hasViewName(viewName)
+                hasErrorType(error.javaClass.name)
+            }
+            .hasRumEvent(index = 2) {
+                hasService(stubSdkCore.getDatadogContext().service)
+                hasApplicationId(fakeApplicationId)
+                hasSessionType("user")
+                hasSource("android")
+                hasType("view")
+                hasViewName(viewName)
+                hasResourceCount(0)
+                hasErrorCount(1)
+            }
+    }
+
+    companion object {
+        private val mainLooper = MainLooperTestConfiguration()
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏ Same here, more information on this please.

Reply:
This is used to ensure we can initialise the RUM feature in a JVM environement (where there's no real Main Looper)